backend/models/time.py

```diff
@@ -194,43 +212,43 @@ def _get_matrix(self, source_points, destination_points):
-        logger.debug('_get_matrix from {} to {} points'.format(
-            source_points[0],
-            len(destination_points)
-        ))
-        request_url = 'https://api.mapbox.com/directions-matrix/v1/mapbox/driving/{points}'
-        all_points = source_points + destination_points
-        if len(all_points) > self.max_matrix_size:
-            raise RuntimeError('Mapbox matrix API only accepts 25 locations.')
-        coordinates = ';'.join([self._represent_point_as_str(point) for point in all_points])
+        return self._get_matrix_http(source_points, destination_points)
 
-        params = {
-            'sources': ';'.join([str(i) for i, _ in enumerate(source_points)]),
-            'destinations': ';'.join(
-                [str(len(source_points) + i) for i, _ in enumerate(destination_points)]
-            ),
-            'access_token': self.api_key
-        }
 
-        response = requests.get(
-            url=request_url.format(points=coordinates),
-            params=params
-        )
+class MapBoxWalkingTime(APITime):
+    """Class for MapBox walking time measurements."""
 
-        response.raise_for_status()
-        content = response.json()
-        return content['durations']
+    def __init__(self, *args, **kwargs):
+        """Initialize the distance class with a MapBox API access token."""
+        logger.debug('Initializing MapBox measurer.')
+        super(MapBoxWalkingTime, self).__init__(*args, **kwargs)
+        self.access_token = self.access_token or os.getenv('MAPBOX_TOKEN')
+        self.api_url = 'https://api.mapbox.com/directions-matrix/v1/mapbox/walking/'
+        # The MapBox API accepts 25 points max as input.
+        self.max_matrix_size = 25
+
+    @rate_limited(1, 1.0)  # 1 per second - period=1, every=1.0.
+    def _get_matrix(self, source_points, destination_points):
+        """
+        Retrieve a time matrix using the MapBox API.
+
+        Time is expressed in seconds.
+        NOTES:
+         - MapBox API requires a list of lon,lat;lon,lat points.
+         - The MapBox API accepts 25 points max as input.
+        """
+        return self._get_matrix_http(source_points, destination_points)
 
 
-class OpenRouteDrivingTime(APIDrivingTime):
+class OpenRouteDrivingTime(APITime):
     """Class for OpenRouteService driving time measurements."""
 
     def __init__(self, *args, **kwargs):
         """Initialize the distance class with an Open Routing Service API key."""
         logger.debug('Initializing OpenRouting measurer.')
         super(OpenRouteDrivingTime, self).__init__(*args, **kwargs)
-        self.api_key = self.api_key or os.getenv('ORS_TOKEN')
-        self.client = openrouteservice.Client(key=self.api_key)
+        self.access_token = self.access_token or os.getenv('ORS_TOKEN')
+        self.client = openrouteservice.Client(key=self.access_token)
 
     @retry(wait_fixed=500, stop_max_attempt_number=2)
     @rate_limited(40, 60.0)  # 40 per minute - period=40, every=60.0.
```

Comment:
So now this class has a method `_get_matrix_http` that raises a `Key/Value/IndexError` whenever it is called, in addition to a method `_get_matrix` that actually uses HTTP under the hood?

Reply:
It does not get called by the parent functions though